### ISSUE TITLE
pow grad has surprising behavior based on dtype

### ISSUE THREAD

**Issue by mattjj** [google/jax#17995]
> Hi @mattjj, thanks for this. I'm finding this choice very confusing, since at the back of my head, I'm dealing with a function in one variable, x, and so it's unexpected that for every expression I write down inside such a function, I need to check to make sure what I think are constants are well-behaved if they were to become variables. But perhaps there are good reasons why this should be. 
> 
> I'm still running into trouble along these lines though, I think when `arange` is used to build the polynomial. Here is a simple example that breaks: 
```
import jax
import jax.numpy as jnp

b = lambda z: jnp.sum(z**jnp.arange(0, 2))
d = lambda z: 1. + z
print(b(2.543), d(2.543)) # check they are the same and I'm not crazy
print(jax.grad(d)(1.))  # gives 1. as expected
print(jax.grad(b)(1.)) # gives 2.
```

_Originally posted by @hongwanliu in https://github.com/google/jax/issues/14397#issuecomment-1703838855_

**Comment by mattjj** [comment]
Copying a comment from #14397 

> If you change the `jnp.arange` to use float dtype, then things agree:
> 
> ```python
> b = lambda z: jnp.sum(z**jnp.arange(0, 2, dtype=float))
> d = lambda z: 1 + z
> 
> print(jax.grad(d)(1.))  # gives 1. as expected
> print(jax.grad(b)(1.)) # gives 1. also
> ```
> 
> I'd say that's pretty surprising though. We probably need to revise this; either there's some bug, or this dtype-based-resolving-of-ambiguities is too subtle and we should raise an error instead (asking the user to be explicit).

**Comment by mattjj** [comment]
Wow, this was just a bug in the JVP rule, and the test I added had the same bug. Sorry! This is what I get for context switching on that PR so many times.

**Comment by mattjj** [comment]
Thanks @hongwanliu for spotting this. If you ever find a numerically incorrect gradient in JAX, please let us know; it's the highest priority to fix.

**Comment by hongwanliu** [comment]
Awesome. Thanks @mattjj for fixing!
